cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/SwitchesChain.java

```diff
@@ -31,232 +29,173 @@
-
-        private final Map<String, List<String>> adjacencies;
+    private static Terminal uniqueTerminalAssociatedConductingEquipmentNodeBreaker(VoltageLevel vl, List<Integer> nodes) {
+        List<Terminal> terminals = new ArrayList<>();
 
-        Adjacency(VoltageLevel vl, String switchId, boolean isNodeBreakerModel) {
-            adjacencies = new HashMap<>();
-
-            if (isNodeBreakerModel) {
-                adjacencyNodeBreaker(vl, switchId, adjacencies);
-            } else {
-                adjacencyBusBreaker(vl, switchId, adjacencies);
+        vl.getConnectables().forEach(c -> {
+            if (isDiscarded(c)) {
+                return;
             }
-        }
 
-        private static void adjacencyNodeBreaker(VoltageLevel vl, String switchId, Map<String, List<String>> adjacency) {
-            vl.getSwitches().forEach(sw -> {
-                if (sw.getId().equals(switchId)) {
+            c.getTerminals().forEach(terminal -> {
+                if (!isSameVoltageLevel((Terminal) terminal, vl)
+                    || !isTerminalInNodes((Terminal) terminal, nodes)) {
                     return;
                 }
-                addAdjacency(adjacency, vl.getNodeBreakerView().getNode1(sw.getId()), vl.getNodeBreakerView().getNode2(sw.getId()));
+                terminals.add((Terminal) terminal);
             });
-            vl.getNodeBreakerView().getInternalConnections().forEach(ic -> addAdjacency(adjacency, ic.getNode1(), ic.getNode2()));
-        }
+        });
 
-        private static void addAdjacency(Map<String, List<String>> adjacency, int nodeEnd1, int nodeEnd2) {
-            String end1 = String.valueOf(nodeEnd1);
-            String end2 = String.valueOf(nodeEnd2);
-            adjacency.computeIfAbsent(end1, k -> new ArrayList<>()).add(end2);
-            adjacency.computeIfAbsent(end2, k -> new ArrayList<>()).add(end1);
+        if (terminals.size() == 1) {
+            return terminals.get(0);
         }
+        return null;
+    }
+
+    private static Terminal uniqueTerminalAssociatedConductingEquipmentBusBranch(VoltageLevel vl, List<String> configuredBuses) {
+        List<Terminal> terminals = new ArrayList<>();
+
+        vl.getConnectables().forEach(c -> {
+            if (isDiscarded(c)) {
+                return;
+            }
 
-        private static void adjacencyBusBreaker(VoltageLevel vl, String switchId, Map<String, List<String>> adjacency) {
-            vl.getSwitches().forEach(sw -> {
-                if (sw.getId().equals(switchId)) {
+            c.getTerminals().forEach(terminal -> {
+                if (!isSameVoltageLevel((Terminal) terminal, vl)
+                    || !isTerminalInConfiguredBuses((Terminal) terminal, configuredBuses)) {
                     return;
                 }
-                addAdjacency(adjacency, vl.getBusBreakerView().getBus1(sw.getId()), vl.getBusBreakerView().getBus2(sw.getId()));
+                terminals.add((Terminal) terminal);
```

Comment:
Rather than returning if (!a || !b), add the terminal only if (a && b). That could be done in a lambda then.

Reply:
Done

Reply:
I simplified what you changed to have a simpler lambda like I suggested

Reply:
ok